File: solutions/fsharp/acronym/1/Acronym.fs
module Acronym
open System
let abbreviate (phrase:string) =
    phrase.Split([|' ';'-';'_'|])
    |> Array.map (fun s -> if not <| String.IsNullOrEmpty s then Char.ToUpper s.[0] |> Some else None)
    |> Array.choose id
    |> System.String.Concat

File: solutions/erlang/acronym/1/src/acronym.erl
-module(acronym).
-export([abbreviate/1]).
abbreviate(Phrase) ->
	parse(Phrase, true, "").
parse([$\s|Str], _, Acc) ->
	parse(Str, true, Acc);
parse([$-|Str], _, Acc) ->
	parse(Str, true, Acc);
parse([C|Str], true, Acc) when C>=$A, C=<$Z ->
	parse(Str, false, [C|Acc]);
parse([C|Str], true, Acc) when C>=$a, C=<$z ->
	parse(Str, false, [C-$a+$A|Acc]);
parse([_|Str], ExpectChar, Acc) ->
	parse(Str, ExpectChar, Acc);
parse([], _, Acc) ->
	lists:reverse(Acc).